### PR TITLE
Dev/lostern/grid9 1

### PR DESCRIPTION
**Pull request by lostern**
NvidiaGPU changes for GRID9.1

### CODE REVIEW

**Comment by vermagit** [comment]
New driver is at the top as array index 0.
DirLink is sort of the true source of the file. FwLink is what the code uses internally since it allows us the freedom to change FwLinks (Externally) to point wherever we want. Hence, the requirement to have FwLink for all versions, even if it is really a download.microsoft.link and not necessarily a 'fwlink'.
As mentioned in the comment on the top of the file, once a new file/driver is added, the FwLink corresponding to this should be updated to point to this new file. This reminds me that I should get you permissions to the FwLinks which need to be kept updated. Working on that now ...

File: NvidiaGPU/resources.json [hunk 1]
```diff
@@ -30,13 +30,21 @@
...
+                  "FwLink" : "http://download.microsoft.com/download/8/C/C/8CC88D54-EB07-44D3-8FA9-B797B173ED04/431.02_grid_win10_server2016_server2019_64bit_international.exe"
+                },
+                {
+                  "Num" : "425.31",
+                  "FwLink" : "http://download.microsoft.com/download/4/8/C/48C2D46E-EB64-460E-A8D9-0F55737D0D68/425.31_grid_win10_server2016_64bit_international.exe"
+                },
                 {
                   "Num" : "412.16",
-                  "DirLink" : "http://download.microsoft.com/download/0/B/0/0B082E50-4162-4910-B422-81B2E06E28B5/412.16_grid_win10_server2016_64bit_international.exe"
+                  "FwLink" : "http://download.microsoft.com/download/0/B/0/0B082E50-4162-4910-B422-81B2E06E28B5/412.16_grid_win10_server2016_64bit_international.exe"
```
Comment (AnnaDaly) [inline NvidiaGPU/resources.json]: this is DirLink

File: NvidiaGPU/resources.json [hunk 2]
```diff
@@ -85,13 +93,21 @@
...
+                  "FwLink" : "http://download.microsoft.com/download/B/D/6/BD6D1C34-E41F-4654-B5AB-F4DA9C08AA64/431.02_grid_win7_win8_server2008R2_server2012R2_64bit_international.exe"
+                },
+                {
+                  "Num" : "425.31",
+                  "FwLink" : "http://download.microsoft.com/download/6/D/7/6D73C628-B5FB-4243-9520-DAEF363223CB/425.31_grid_win8_win7_server2012R2_server2008R2_64bit_international.exe"
+                },
                 {
                   "Num" : "412.16",
-                  "DirLink" : "http://download.microsoft.com/download/0/9/8/098B5E8A-0563-4B25-9CDB-173B03B52C60/412.16_grid_win8_win7_server2012R2_server2008R2_64bit_international.exe"
+                  "FwLink" : "http://download.microsoft.com/download/0/9/8/098B5E8A-0563-4B25-9CDB-173B03B52C60/412.16_grid_win8_win7_server2012R2_server2008R2_64bit_international.exe"
```
Comment (AnnaDaly) [inline NvidiaGPU/resources.json]: same, this is dirlinks

File: NvidiaGPU/resources.json [hunk 3]
```diff
@@ -30,13 +30,21 @@
               "Type" : "GRID",
               "Version" : [
                 {
-                  "Num" : "425.31",
-                  "DirLink" : "http://download.microsoft.com/download/4/8/C/48C2D46E-EB64-460E-A8D9-0F55737D0D68/425.31_grid_win10_server2016_64bit_international.exe",
+                  "Num" : "431.79",
+                  "DirLink" : "https://download.microsoft.com/download/e/8/2/e8257939-a439-4da8-a927-b64b63743db1/431.79_grid_win10_server2016_server2019_64bit_international.exe",
                   "FwLink" : "https://go.microsoft.com/fwlink/?linkid=874181"
```
Comment (AnnaDaly) [inline NvidiaGPU/resources.json]: please make sure to update fwlinks

File: NvidiaGPU/resources.json (hunk 2 above)
Comment (lostern) [reply NvidiaGPU/resources.json]: FwLinks are being used according to Aman

File: NvidiaGPU/resources.json (hunk 3 above)
Comment (lostern) [reply NvidiaGPU/resources.json]: Fwlinks updated

**Pull request merged**